api/v1alpha1/systemconfiguration_types.go

```diff
@@ -23,6 +23,13 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+// StorageNodeType defines a type of storage node
+type StorageNodeType string
+
+const (
+	Rabbit StorageNodeType = "Rabbit"
```

Comment:
This feels wrong.  Enums are nice, to get the api server to do some validation for you.  In this case, why does the type field need to be an enum?

Reply:
"Rabbit" is already in the CRD's enum list, I'm just making it publicly available in Go so users (nnf-deploy, nnf-dm) can use the definition when creating the config or accessing it.